### ISSUE TITLE
Separate charts into another plugin

### ISSUE THREAD

**Issue by manland** [manland/mattermost-plugin-analytics#14]
It could serve to another purpose. Check if plugin discovery is possible.

**Comment by hanzei** [comment]
Could you elaborate a bit? Do you want to move the "chart serving" into another plugin?

**Comment by manland** [comment]
@hanzei yes exactly, I think of a Prometheus plugin or a chat renderer when you upload CSV files...

I can make a separate plugin and force user to install it, or make a special package and use it in other plugin (more like a lib)

Any thought about that?

**Comment by hanzei** [comment]
Maybe I'm not getting your point. But why would you do this? These charts are specific to this plugin.

**Comment by manland** [comment]
These charts are not specific. I build an url containing all data necessary to draw the chart : `![users bar chart](http://127.0.0.1:8065/plugins/com.github.manland.mattermost-plugin-analytics/bar.svg?manland=3&sysadmin=37&user-1=3)` I can change `manland=3&sysadmin=37&user-1=3` with anything I want to draw (key=value) for bar chart. Its the same for pie and line charts.

**Comment by hanzei** [comment]
Ah, I see. Yes, that would make sense then.

Forcing the installation of another plugin is not strait forward but deficiently possible.

**Comment by hanzei** [comment]
I would like to look into this, once I have the time :man_shrugging:

**Comment by manland** [comment]
> Only one life, so many things to do...

I know, I know :D

And yes I think a lib for plugins is better than an other plugin. In user perspective is not easy to understand the need of install 2 plugins.

**Comment by hanzei** [comment]
I would expect this plugin taking care of installing all needed plugins.

**Comment by manland** [comment]
Oo wouhaou i didn't see this like that :D Interesting! I will take a look at this.

**Comment by hanzei** [comment]
Just call [`GetPlugins](https://developers.mattermost.com/extend/plugins/server/reference/#API.GetPlugins) and check the result. You could even offer the system admin to install the plugin for him.
